protocols/request-response/src/lib.rs

```diff
@@ -285,29 +285,51 @@ impl fmt::Display for RequestId {
 pub struct Config {
     request_timeout: Duration,
     connection_keep_alive: Duration,
+    max_concurrent_streams: usize,
 }
 
 impl Default for Config {
     fn default() -> Self {
         Self {
             connection_keep_alive: Duration::from_secs(10),
             request_timeout: Duration::from_secs(10),
+            max_concurrent_streams: 100,
         }
     }
 }
 
 impl Config {
     /// Sets the keep-alive timeout of idle connections.
+    #[deprecated(note = "Use `Config::with_connection_keep_alive` for one-liner constructions.")]
     pub fn set_connection_keep_alive(&mut self, v: Duration) -> &mut Self {
         self.connection_keep_alive = v;
         self
     }
 
     /// Sets the timeout for inbound and outbound requests.
+    #[deprecated(note = "Use `Config::with_request_timeout` for one-liner constructions.")]
```

Comment:
@mxinden Would you agree that we can publish this as a patch release if we link to an issue / migration document here that explains that we are now capping streams at 100?